src/main/java/com/chargerrobotics/sensors/GyroscopeSerial.java

```diff
@@ -1,65 +1,88 @@
 package com.chargerrobotics.sensors;
 
-import java.nio.ByteBuffer;
-
 import com.chargerrobotics.utils.ArduinoSerial;
 import com.chargerrobotics.utils.ArduinoSerialReceiver;
 import com.chargerrobotics.utils.ArduinoSerialReceiver.ArduinoListener;
+import java.nio.ByteBuffer;
 
 public class GyroscopeSerial extends ArduinoListener {
 
-	private volatile float x = 0;
-	private volatile float y = 0;
-	private volatile float z = 0;
-	
-	public GyroscopeSerial() {
-		ArduinoSerialReceiver.registerListener(this, (short)0xA02D);
-	}
+  private volatile float x = 0;
+  private volatile float y = 0;
+  private volatile float z = 0;
+
+  public GyroscopeSerial() {
+    ArduinoSerialReceiver.registerListener(this, (short) 0xA02D);
+  }
+
+  public void receiveData(ArduinoSerial serial, ByteBuffer buffer) {
+    if (buffer.remaining() >= 12) {
+      x = buffer.getFloat();
+      y = buffer.getFloat();
+      z = buffer.getFloat();
+    }
+  }
+
+  /**
+   * Gets the robot heading in degrees or {@code -1} if data is expired
+   *
+   * <p>Equivalent to {@link #getYaw()}
+   *
+   * @return Robot heading
+   */
+  public float getHeading() {
+    return getYaw();
+  }
+
+  /**
+   * Gets the robot heading in radians on the interval [-pi, pi) or {@code -1} if data is expired.
+   * This method is counterclockwise-positive, but the gyro is clockwise positive
+   *
+   * @return Robot heading radians
+   */
+  public float getHeadingRadians() {
+    float headingRad = x;
+    if (headingRad > 180) {
+      headingRad = 360 - x;
+    } else {
+      headingRad = -x;
+    }
+    return isExpired() ? -1 : (float) Math.toRadians(headingRad);
+  }
+
+  /**
+   * Gets the robot yaw in degrees or {@code -1} if data is expired
```

Comment:
Ditto here if this can also return `-1` as a valid value.

Reply:
Not necessary here as the gyro returns the heading in degrees with counter-clockwise being positive. Also, the values roll over from 359 back to 0. This was discovered in #3.